File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/TmModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.Moves;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class TmModifierItem extends PokemonModifierItem {

    public static final String TARGET = "TmModifierType";

    private Moves moveId;

}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/TmModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.Moves

class TmModifierItem : PokemonModifierItem() {

    companion object {
        const val TARGET = "TmModifierType"
    }

    var moveId: Moves? = null
}